### ISSUE TITLE
getting error when set eval_acc = True

### ISSUE THREAD

**Issue by rnyak** [IndicoDataSolutions/finetune#412]
**Describe the bug**
I am using Classifier.

I wanted to add eval_acc param. This is my  command line:

`model = Classifier(n_epochs=2, base_model=GPT2Model, tensorboard_folder ='/workspace/checkpoints', max_length= 1024, val_size = 100, chunk_long_sequences=False, keep_best_model= True, params_device =0, eval_acc = True)
model.fit(trainX, trainY)`

I am getting this error:
```
INFO:finetune:Adding evaluation metrics, Accuracy
---------------------------------------------------------------------------
InvalidArgumentError                      Traceback (most recent call last)
/usr/local/lib/python3.6/dist-packages/tensorflow/python/framework/ops.py in _create_c_op(graph, node_def, inputs, control_inputs)
   1863   try:
-> 1864     c_op = c_api.TF_FinishOperation(op_desc)
   1865   except errors.InvalidArgumentError as e:

InvalidArgumentError: Can not squeeze dim[1], expected a dimension of 1, got 2 for 'remove_squeezable_dimensions/Squeeze' (op: 'Squeeze') with input shapes: [?,2].

During handling of the above exception, another exception occurred:

ValueError                                Traceback (most recent call last)
<ipython-input-71-940779edda1d> in <module>
      3 start = time.time()
      4 model = Classifier(n_epochs=2, base_model=GPT2Model, tensorboard_folder ='/workspace/checkpoints', max_length= 1024, val_size = 100, chunk_long_sequences=False, keep_best_model= True, params_device =0, eval_acc = True)
----> 5 model.fit(trainX, trainY)
      6 time.time() - start

/Finetune/finetune/base.py in fit(self, *args, **kwargs)
    453     def fit(self, *args, **kwargs):
    454         """ An alias for finetune. """
--> 455         return self.finetune(*args, **kwargs)
    456 
    457     def _predict(self, Xs):

/Finetune/finetune/target_models/classifier.py in finetune(self, X, Y, batch_size)
    121                            corresponds to the number of training examples provided to each GPU.
    122         """
--> 123         return super().finetune(X, Y=Y, batch_size=batch_size)
    124 
    125     @classmethod

/Finetune/finetune/base.py in finetune(self, Xs, Y, batch_size)
    263 
    264                 tf.logging.info("Finishing pre-fit initialisation...")
--> 265             estimator.train(train_input_fn, hooks=train_hooks, steps=num_steps)
    266 
    267         self._trained = True

/usr/local/lib/python3.6/dist-packages/tensorflow_estimator/python/estimator/estimator.py in train(self, input_fn, hooks, steps, max_steps, saving_listeners)
    365 
    366       saving_listeners = _check_listeners_type(saving_listeners)
--> 367       loss = self._train_model(input_fn, hooks, saving_listeners)
    368       logging.info('Loss for final step: %s.', loss)
    369       return self

/usr/local/lib/python3.6/dist-packages/tensorflow_estimator/python/estimator/estimator.py in _train_model(self, input_fn, hooks, saving_listeners)
   1154   def _train_model(self, input_fn, hooks, saving_listeners):
   1155     if self._train_distribution:
-> 1156       return self._train_model_distributed(input_fn, hooks, saving_listeners)
   1157     else:
   1158       return self._train_model_default(input_fn, hooks, saving_listeners)

/usr/local/lib/python3.6/dist-packages/tensorflow_estimator/python/estimator/estimator.py in _train_model_distributed(self, input_fn, hooks, saving_listeners)
   1217       self._config._train_distribute.configure(self._config.session_config)
   1218       return self._actual_train_model_distributed(
-> 1219           self._config._train_distribute, input_fn, hooks, saving_listeners)
   1220     # pylint: enable=protected-access
   1221 

/usr/local/lib/python3.6/dist-packages/tensorflow_estimator/python/estimator/estimator.py in _actual_train_model_distributed(self, strategy, input_fn, hooks, saving_listeners)
   1327         return self._train_with_estimator_spec(estimator_spec, worker_hooks,
   1328                                                hooks, global_step_tensor,
-> 1329                                                saving_listeners)
   1330 
   1331   def _train_with_estimator_spec_distributed(self, estimator_spec, worker_hooks,

/usr/local/lib/python3.6/dist-packages/tensorflow_estimator/python/estimator/estimator.py in _train_with_estimator_spec(self, estimator_spec, worker_hooks, hooks, global_step_tensor, saving_listeners)
   1478         save_summaries_steps=save_summary_steps,
   1479         config=self._session_config,
-> 1480         log_step_count_steps=log_step_count_steps) as mon_sess:
   1481       loss = None
   1482       any_step_done = False

/usr/local/lib/python3.6/dist-packages/tensorflow/python/training/monitored_session.py in MonitoredTrainingSession(master, is_chief, checkpoint_dir, scaffold, hooks, chief_only_hooks, save_checkpoint_secs, save_summaries_steps, save_summaries_secs, config, stop_grace_period_secs, log_step_count_steps, max_wait_secs, save_checkpoint_steps, summary_dir)
    582       session_creator=session_creator,
    583       hooks=all_hooks,
--> 584       stop_grace_period_secs=stop_grace_period_secs)
    585 
    586 

/usr/local/lib/python3.6/dist-packages/tensorflow/python/training/monitored_session.py in __init__(self, session_creator, hooks, stop_grace_period_secs)
   1005         hooks,
   1006         should_recover=True,
-> 1007         stop_grace_period_secs=stop_grace_period_secs)
   1008 
   1009 

/usr/local/lib/python3.6/dist-packages/tensorflow/python/training/monitored_session.py in __init__(self, session_creator, hooks, should_recover, stop_grace_period_secs)
    711     self._hooks = hooks or []
    712     for h in self._hooks:
--> 713       h.begin()
    714 
    715     worker_context = distribute_coordinator_context.get_current_worker_context()

/usr/local/lib/python3.6/dist-packages/tensorflow_estimator/python/estimator/hooks/hooks.py in begin(self)
    134       (self._scaffold, self._update_op, self._eval_dict,
    135        self._all_hooks) = self._estimator._evaluate_build_graph(
--> 136            self._input_fn, self._hooks, checkpoint_path=None)
    137 
    138       if self._scaffold.saver is not None:

/usr/local/lib/python3.6/dist-packages/tensorflow_estimator/python/estimator/estimator.py in _evaluate_build_graph(self, input_fn, hooks, checkpoint_path)
   1499     else:
   1500       (scaffold, evaluation_hooks, input_hooks, update_op, eval_dict) = (
-> 1501           self._call_model_fn_eval(input_fn, self.config))
   1502 
   1503     global_step_tensor = training_util.get_global_step(ops.get_default_graph())

/usr/local/lib/python3.6/dist-packages/tensorflow_estimator/python/estimator/estimator.py in _call_model_fn_eval(self, input_fn, config)
   1535 
   1536     estimator_spec = self._call_model_fn(
-> 1537         features, labels, ModeKeys.EVAL, config)
   1538     eval_metric_ops = _verify_and_create_loss_metric(
   1539         estimator_spec.eval_metric_ops, estimator_spec.loss)

/usr/local/lib/python3.6/dist-packages/tensorflow_estimator/python/estimator/estimator.py in _call_model_fn(self, features, labels, mode, config)
   1144 
   1145     logging.info('Calling model_fn.')
-> 1146     model_fn_results = self._model_fn(features=features, **kwargs)
   1147     logging.info('Done calling model_fn.')
   1148 

/Finetune/finetune/model.py in _model_fn(features, labels, mode, params)
    290             LOGGER.info("Adding evaluation metrics, Accuracy")
    291             labels_dense = tf.argmax(labels, -1)
--> 292             metrics = {"Accuracy": tf.metrics.accuracy(pred_op, labels_dense)}
    293         else:
    294             metrics = None

/usr/local/lib/python3.6/dist-packages/tensorflow/python/ops/metrics_impl.py in accuracy(labels, predictions, weights, metrics_collections, updates_collections, name)
    450 
    451   predictions, labels, weights = _remove_squeezable_dimensions(
--> 452       predictions=predictions, labels=labels, weights=weights)
    453   predictions.get_shape().assert_is_compatible_with(labels.get_shape())
    454   if labels.dtype != predictions.dtype:

/usr/local/lib/python3.6/dist-packages/tensorflow/python/ops/metrics_impl.py in _remove_squeezable_dimensions(predictions, labels, weights)
    113   if labels is not None:
    114     labels, predictions = confusion_matrix.remove_squeezable_dimensions(
--> 115         labels, predictions)
    116     predictions.get_shape().assert_is_compatible_with(labels.get_shape())
    117 

/usr/local/lib/python3.6/dist-packages/tensorflow/python/ops/confusion_matrix.py in remove_squeezable_dimensions(labels, predictions, expected_rank_diff, name)
     71         predictions = array_ops.squeeze(predictions, [-1])
     72       elif rank_diff == expected_rank_diff - 1:
---> 73         labels = array_ops.squeeze(labels, [-1])
     74       return labels, predictions
     75 

/usr/local/lib/python3.6/dist-packages/tensorflow/python/util/dispatch.py in wrapper(*args, **kwargs)
    178     """Call target, and fall back on dispatchers if there is a TypeError."""
    179     try:
--> 180       return target(*args, **kwargs)
    181     except (TypeError, ValueError):
    182       # Note: convert_to_eager_tensor currently raises a ValueError, not a

/usr/local/lib/python3.6/dist-packages/tensorflow/python/util/deprecation.py in new_func(*args, **kwargs)
    505                 'in a future version' if date is None else ('after %s' % date),
    506                 instructions)
--> 507       return func(*args, **kwargs)
    508 
    509     doc = _add_deprecated_arg_notice_to_docstring(

/usr/local/lib/python3.6/dist-packages/tensorflow/python/ops/array_ops.py in squeeze(input, axis, name, squeeze_dims)
   3160   if np.isscalar(axis):
   3161     axis = [axis]
-> 3162   return gen_array_ops.squeeze(input, axis, name)
   3163 
   3164 

/usr/local/lib/python3.6/dist-packages/tensorflow/python/ops/gen_array_ops.py in squeeze(input, axis, name)
   9664   axis = [_execute.make_int(_i, "axis") for _i in axis]
   9665   _, _, _op = _op_def_lib._apply_op_helper(
-> 9666         "Squeeze", input=input, squeeze_dims=axis, name=name)
   9667   _result = _op.outputs[:]
   9668   _inputs_flat = _op.inputs

/usr/local/lib/python3.6/dist-packages/tensorflow/python/framework/op_def_library.py in _apply_op_helper(self, op_type_name, name, **keywords)
    786         op = g.create_op(op_type_name, inputs, dtypes=None, name=scope,
    787                          input_types=input_types, attrs=attr_protos,
--> 788                          op_def=op_def)
    789       return output_structure, op_def.is_stateful, op
    790 

/usr/local/lib/python3.6/dist-packages/tensorflow/python/util/deprecation.py in new_func(*args, **kwargs)
    505                 'in a future version' if date is None else ('after %s' % date),
    506                 instructions)
--> 507       return func(*args, **kwargs)
    508 
    509     doc = _add_deprecated_arg_notice_to_docstring(

/usr/local/lib/python3.6/dist-packages/tensorflow/python/framework/ops.py in create_op(***failed resolving arguments***)
   3614           input_types=input_types,
   3615           original_op=self._default_original_op,
-> 3616           op_def=op_def)
   3617       self._create_op_helper(ret, compute_device=compute_device)
   3618     return ret

/usr/local/lib/python3.6/dist-packages/tensorflow/python/framework/ops.py in __init__(self, node_def, g, inputs, output_types, control_inputs, input_types, original_op, op_def)
   2025           op_def, inputs, node_def.attr)
   2026       self._c_op = _create_c_op(self._graph, node_def, grouped_inputs,
-> 2027                                 control_input_ops)
   2028 
   2029     # Initialize self._outputs.

/usr/local/lib/python3.6/dist-packages/tensorflow/python/framework/ops.py in _create_c_op(graph, node_def, inputs, control_inputs)
   1865   except errors.InvalidArgumentError as e:
   1866     # Convert to ValueError for backwards compatibility.
-> 1867     raise ValueError(str(e))
   1868 
   1869   return c_op

ValueError: Can not squeeze dim[1], expected a dimension of 1, got 2 for 'remove_squeezable_dimensions/Squeeze' (op: 'Squeeze') with input shapes: [?,2].
```

It works without issue when I remove eval_acc = True.

**Minimum Reproducible Example**
A short code snippet which reproduces the exception

**Expected behavior**

**Additional context**

**Comment by benleetownsend** [comment]
Thanks for the bug report. This is fixed in #370 but this isn't merged yet. 

I have isolated the patch for this bug and put it in a branch `ben/fix_eval_acc` for you to work from if you require a fix immediately.

This will be included in the next pypy release but we do not currently have a timeline for this.

**Comment by madisonmay** [comment]
@rnyak 
0.8.4 (with a fix for your issue) is now live.
